envsetup.sh

```diff
@@ -297,6 +297,33 @@ function oe_search_text() {
   cd -
 }
 
+function oe_add_layer() {
+  if test -z $1; then
+    echo "Usage:  oe_add_layer <git URI> [<branch>]"
+    return
+  fi
+  cd $OE_BASE
+  if [ -z "$2" ]; then
+    br="master"
+  else
+    br="$2"
+  fi
+  n=`echo $1 | awk -F "[/:]" '{ print $NF }'`
+  git submodule add -b $br -f $1 sources/$n \
+  && bitbake-layers add-layer sources/$n \
+  && sed -i -e "s|$OE_BASE|\${TOPDIR}|" conf/bblayers.conf
+}
+
+function oe_remove_layer() {
+  if test -z $1; then
+    echo "Usage:  oe_remove_layer <layer-name>"
+    return
+  fi
+  cd $OE_BASE
+  bitbake-layers remove-layer $1 \
+  && git rm -f sources/$1
```

Comment:
it would be very nice to have this automated, as I have to look up how to remove submodules every time I do it. Lately I have been doing:

git submodule deinit <sources/layer>
git rm sources/layer

Not sure it that cleans up completely, but seems to work.